transport/peers/peer.go

```diff
@@ -0,0 +1,82 @@
+package peers
+
+import (
+	"go.uber.org/yarpc/transport"
+
+	"github.com/uber-go/atomic"
+)
+
+// NewPeerIdentifier creates a new HostPortPeerIdentifier
+func NewPeerIdentifier(hostport string) *HostPortPeerIdentifier {
+	return &HostPortPeerIdentifier{
```

Comment:
I would pass this around by value. You're not getting much by making it a
pointer. It's a very light object. You can also just make it an alias.

```
type HostPortPeerIdentifier string

func (hp HostPortPeerIdentifier) String() string {
    return string(hp)
}
```